src/ecs/resources/input.rs

```diff
@@ -15,27 +15,65 @@ enum KeyQueryState {
 }
 
 /// An iterator over all currently pressed down keys.
-pub struct PressedKeysIterator<'a> {
+pub struct PressedKeys<'a> {
     iterator: Keys<'a, VirtualKeyCode, KeyQueryState>,
 }
 
-impl<'a> Iterator for PressedKeysIterator<'a> {
+impl<'a> Iterator for PressedKeys<'a> {
     type Item = &'a VirtualKeyCode;
     fn next(&mut self) -> Option<Self::Item> {
         self.iterator.next()
     }
 }
 
-/// Processes user input events.
+/// An iterator over all currently pressed down mouse buttons.
+pub struct PressedMouseButtons<'a> {
+    iterator: Keys<'a, MouseButton, KeyQueryState>,
+}
+
+impl<'a> Iterator for PressedMouseButtons<'a> {
+    type Item = &'a MouseButton;
+    fn next(&mut self) -> Option<Self::Item> {
+        self.iterator.next()
+    }
+}
+
+/// This struct holds state information about input devices.
+///
+/// For example, if a key is pressed on the keyboard, this struct will record
+/// that the key is pressed until it is released again. Usage requires pumping input events through
+/// this struct, using the following code in the [handle_events] method of [State].
+///
+/// [State]: ../../../engine/trait.State.html
+/// [handle_events]: ../../../engine/trait.State.html#method.handle_events
+///
+/// ```ignore
+/// fn handle_events(&mut self,
+///                  events: &[WindowEvent],
+///                  _: &mut World,
+///                  _: &mut AssetManager,
+///                  _: &mut Pipeline)
+///                  -> Trans {
+///     // ...
+///     let mut input_handler = world.write_resource::<InputHandler>();
+///     input_handler.update(events);
+///     // ...
+/// }
+/// ```
 #[derive(Default)]
 pub struct InputHandler {
     pressed_keys: HashMap<VirtualKeyCode, KeyQueryState>,
+    // use KeyQueryState for mouse button query state as well
```

Comment:
I just wanted to emphaize that even though the name `KeyQueryState` is specific to keys, its structure is appropriate for mouse buttons as well. However, I'm happy to remove.